File: sdks/php/examples/quickstart.php
<?php

declare(strict_types=1);

// Quickstart tour of the PHP SDK.
//   BACKLEX_URL=http://localhost:5173 BACKLEX_KEY=pak_... php examples/quickstart.php

require __DIR__ . '/../src/BacklexException.php';
require __DIR__ . '/../src/Filter.php';
require __DIR__ . '/../src/QueryBuilder.php';
require __DIR__ . '/../src/Collection.php';
require __DIR__ . '/../src/Auth.php';
require __DIR__ . '/../src/Storage.php';
require __DIR__ . '/../src/Realtime.php';
require __DIR__ . '/../src/Client.php';

use Backlex\BacklexException;
use Backlex\Client;
use Backlex\Filter as F;

$url = getenv('BACKLEX_URL') ?: 'http://localhost:5173';
$client = new Client($url, ['api_key' => getenv('BACKLEX_KEY') ?: null]);

// Fluent query builder → compiles to canonical JSON (same wire format as every other SDK).
$query = $client->from('posts')->query()
    ->where(F::and_(
        F::eq('published', true),
        F::gte('views', 100),
        F::rel('author', F::eq('tier', 'gold')),
        F::gte('created_at', F::now(sub: ['days' => 7])),
    ))
    ->select('id', 'title', 'author.name')
    ->orderBy('-created_at')
    ->limit(10)
    ->withMeta('filter_count');

try {
    $res = $query->list();
    echo 'got ' . count($res['data']) . " posts\n";
} catch (BacklexException $e) {
    echo "list failed: {$e->status} {$e->code} — {$e->getMessage()}\n";
}

// CRUD
// $created = $client->from('posts')->create(['title' => 'Hello']);

// Realtime (SSE — blocking in PHP; $onEvent returning false stops)
// $client->subscribe('items:posts', function (array $ev): void {
//     echo "event: {$ev['event']}\n";
// });

File: sdks/rust/examples/quickstart.rs
//! Quickstart tour of the Rust SDK.
//!   BACKLEX_URL=http://localhost:5173 BACKLEX_KEY=pak_... cargo run --example quickstart

use backlex::{filter as f, BacklexError, Client};
use serde_json::json;

fn main() {
    let url = std::env::var("BACKLEX_URL").unwrap_or_else(|_| "http://localhost:5173".into());
    let mut builder = Client::builder(url);
    if let Ok(key) = std::env::var("BACKLEX_KEY") {
        builder = builder.api_key(key);
    }
    let client = builder.build();

    // Fluent query builder → compiles to canonical JSON (same wire format as every other SDK).
    let mut sub = serde_json::Map::new();
    sub.insert("days".into(), json!(7));
    let query = client
        .from("posts")
        .query()
        .filter(f::and(vec![
            f::eq("published", json!(true)),
            f::gte("views", json!(100)),
            f::rel("author", vec![f::eq("tier", json!("gold"))]),
            f::gte("created_at", f::now(None, Some(sub))),
        ]))
        .select(&["id", "title", "author.name"])
        .order_by(&["-created_at"])
        .limit(10)
        .with_meta("filter_count");

    match query.list() {
        Ok(res) => {
            let n = res.get("data").and_then(|d| d.as_array()).map(|a| a.len()).unwrap_or(0);
            println!("got {} posts (meta={:?})", n, res.get("meta"));
        }
        Err(BacklexError { status, code, message, .. }) => {
            println!("list failed: {} {} — {}", status, code, message);
        }
    }

    // CRUD
    // let created = client.from("posts").create(&json!({ "title": "Hello" }))?;

    // Realtime (SSE on a thread)
    // let sub = client.subscribe("items:posts", |ev| println!("event: {}", ev["event"]));
    // std::thread::sleep(std::time::Duration::from_secs(5));
    // sub.cancel();
}
